src/python/grpcio_channelz/grpc_channelz/v1/_channelz_page.py

```diff
@@ -0,0 +1,277 @@
+                    _channelz_pb2.GetSocketResponse(),
+                ).socket)
+        servers_n_sockets.append(_SERVER_N_SOCKETS(server, listen_sockets))
+
+    return render(
+        num_servers=len(servers),
+        min_id=min(server.ref.server_id for server in servers),
+        max_id=max(server.ref.server_id for server in servers),
+        servers_n_sockets=servers_n_sockets)
+
+
+def _serversockets_handler(render, args):
+    if 'server_id' not in args:
+        raise _BadRequest('"server_id" cannot be empty')
+    server_id = int(args.get('server_id'))
+    start_socket_id = int(args.get('start_socket_id', '0'))
+    server = json_format.Parse(
+        cygrpc.channelz_get_server(server_id),
+        _channelz_pb2.GetServerResponse(),
+    ).server
+    serversocket_refs = json_format.Parse(
+        cygrpc.channelz_get_server_sockets(server_id, start_socket_id, 0),
+        _channelz_pb2.GetServerSocketsResponse(),
+    ).socket_ref
+
+    if not serversocket_refs:
+        raise _NotFound(
+            'No server socket found for "server_id"==%d' % server_id)
+
+    serversockets = []
+    for ref in serversocket_refs:
+        serversockets.append(
+            json_format.Parse(
+                cygrpc.channelz_get_socket(ref.socket_id),
+                _channelz_pb2.GetSocketResponse(),
+            ).socket)
+
+    return render(
+        server=server,
+        serversockets=serversockets,
+        num_serversockets=len(serversockets),
+        min_id=min(ref.socket_id for ref in serversocket_refs),
+        max_id=max(ref.socket_id for ref in serversocket_refs))
+
+
+_SERVING_PAGES = {
+    '':
+    _PAGE('<nil>', 'index.html', _homepage_handler),
+    'channel':
+    _PAGE('Channel', 'channel.html', _channel_handler),
+    'servers':
+    _PAGE('Servers', 'servers.html', _servers_handler),
+    'serversockets':
+    _PAGE('ServerSockets', 'serversockets.html', _serversockets_handler),
+    'socket':
+    _PAGE('Socket', 'socket.html', _socket_handler),
+    'subchannel':
+    _PAGE('Subchannel', 'subchannel.html', _subchannel_handler),
+    'topchannels':
+    _PAGE('TopChannels', 'topchannels.html', _topchannels_handler),
+}
+
+
+class _RequestHandler(BaseHTTPRequestHandler):
+
+    def _set_ok_headers(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+
+    def _handle(self):
+        if not self.path.startswith('/gdebug/channelz/'):
+            raise _NotFound()
+
+        request_page = self.path[17:].split('?', 1)[0]
+        if request_page not in _SERVING_PAGES:
+            raise _NotFound('Page not found')
+        serving_page = _SERVING_PAGES[request_page]
+
+        base_page = _renderer.format(
+            _base_template,
+            title=serving_page.title,
+            content=_fetch_template(serving_page.template))
+
+        # Encapsulate _renderer and base_page into an enclosure
```

Comment:
Does enclosure=closure? Assuming so, since `_renderer` is not a member variable here, why not just pass `base_page` to `serving_page.handler`?

Reply:
Yes...it should be `closure`. I replaced `render` with `base_page`. I thought it can be less wordy to pass a function named `render`.